package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,228 @@
+    explained_variance : array, (n_components, )
+        Percentage of variance explained by each of the selected components.
+        If a subset of components is not chosen then all components are stored
+        and the sum of explained variances is equal to 1.0.
+
+    Methods
+    -------
+    fit(traj=None, select='All', start=None, stop=None, step=None)
+        Find the principal components of selected atoms in a subset of the
+        trajectory.
+    transform(traj=None, n_components=None)
+        Take the original trajectory and project it onto the principal
+        components.
+    inverse_tranform(pca_space)
+        Take a pca_space and map it back onto the trajectory used to create it.
+    """
+
+    def __init__(self, atomgroup, select='All', demean_ref=None,
+                 n_components=None, **kwargs):
+        """
+        Parameters
+        ----------
+        atomgroup: MDAnalysis atomgroup
+            AtomGroup to be used for PCA.
+        n_components : int, optional
+            The number of principal components to be saved, default saves
+            all principal components, Default: -1
+        start : int, optional
+            First frame of trajectory to use for generation
+            of covariance matrix, Default: None
+        stop : int, optional
+            Last frame of trajectory to use for generation
+            of covariance matrix, Default: None
+        step : int, optional
+            Step between frames of trajectory to use for generation
+            of covariance matrix, Default: None
+        """
+        super(PCA, self).__init__(atomgroup.universe.trajectory,
+                                  **kwargs)
+        self._u = atomgroup.universe
+        # for transform function
+        self.start = kwargs.get('start')
+        self.stop = kwargs.get('stop')
+        self.step = kwargs.get('step')
+
+        if demean_ref:
+            self._demean = True
+            self._reference = demean_ref
+        else:
+            self._demean = False
+
+        self._atoms = atomgroup.select_atoms(select)
+        self.n_components = n_components
+        self._n_atoms = self._atoms.n_atoms
+        self._calculated = False
+
+
+    def _prepare(self):
+        n_dim = self._n_atoms * 3
+        self.cov = np.zeros((n_dim, n_dim))
+        if self._demean:
+            self._ref_atoms = self._reference
+            self._ref_cog = self._ref_atoms.center_of_geometry()
+
+    def _single_frame(self):
+        if self._demean:
+            # TODO make this more functional, initial proof of concept
+            mobile_atoms, old_rmsd = _fit_to(self._atoms.positions,
+                                             self._ref_atoms.positions,
+                                             self._atoms, self._ref_cog,
+                                             self._atoms.center_of_geometry())
+            # now all structures are aligned to reference
+            x = mobile_atoms.positions.ravel()
```

Comment:
move this out of the if-else clause then you can skip the else clause completely. (and remove code duplication)